jobs/mongodb_migration/src/mongodb_migration/migrations/_20230309141600_cache_add_job_runner_version.py

```diff
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+
+from libcommon.simple_cache import CachedResponse
+from mongoengine.connection import get_db
+
+from mongodb_migration.check import check_documents
+from mongodb_migration.migration import Migration
+
+
+# connection already occurred in the main.py (caveat: we use globals)
+class MigrationAddJobRunnerVerionToCacheResponse(Migration):
+    def up(self) -> None:
+        # See https://docs.mongoengine.org/guide/migration.html#example-1-addition-of-a-field
+        logging.info("If missing, add 'job_runner_version' field based on 'worker_version' value")
+        db = get_db("cache")
+        db["cachedResponsesBlue"].update_many(
+            {"job_runner_version": {"$exists": False}},
+            [
+                {
+                    "$set": {
+                        "job_runner_version": {
+                            "$convert": {
+                                "input": {"$first": {"$split": ["$worker_version", "."]}},
```

Comment:
possibly cache entries don't have a worker_version, or it's `null` (we should check). How will it be managed? Should we do two updates? one for non-null worker_version, and another one for non-existent or null worker_version -> default value: null?

Reply:
Oh... it should be handled by onError and onNull, right?

You even test it :) OK!

Reply:
(ignore the comments, the tests show that it's perfectly handled by your migration code!)